### ISSUE TITLE
Unclear code size restriction. Forever `account_uninit`

### ISSUE THREAD

**Issue by qpIlIpp** [ton-blockchain/ton#179]
TL;DR: If code is larger than some unclear size, contracts stop initialize. How can we avoid it or when it will be fixed?

We had started to make some contracts on testnet, but we've met a problem: after some volume of the source code, contract stays in forever `state:account_uninit` because they don't accept set StateInit message.

I want to show a synthetic example.

Let's make a simple smartcontract code: 
```
() recv_internal(int msg_value, cell in_msg_cell, slice in_msg) impure {
}

() recv_external(slice in_msg) impure { 
  accept_message();
  set_data(begin_cell().store_uint(0, 32).store_uint(1, 32).store_uint(2, 32).store_uint(3, 32).store_uint(4, 32).store_uint(5, 32).store_uint(6, 32).store_uint(7, 32).store_uint(8, 32).store_uint(9, 32).end_cell());
}

int m0() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
```

After that, we initiate it with default fif contract creation
```
#!/usr/bin/env fift -s
"TonUtil.fif" include
"Asm.fif" include

"dice.fc.fif" include =: code
0 =: testshift
<b testshift 8 u, b> =: data
null =: libs

<b b{0011} s, code ref, data ref, libs dict, b> =: StateInit

StateInit hashu 0 swap 2dup 2constant code_addr
."new wallet address = " 2dup .addr cr
."Non-bounceable address (for init): " 7 .Addr cr

<b b{1000100} s, code_addr addr, b{000010} s, StateInit <s s, b{0} s, 0 32 u, b> 
2 boc+>B dup Bx. cr
"query.boc" tuck B>file
```
Make transfer to gotten non-bounceable address and send `query.boc` for initiate

For the short contract it works:

```
getaccount 0QCAMv3amOCG8ec3UgSeO6vNebInIA6r2YiG3W34gfZGB-on
...
 state:(account_active
...
```

Now we make the same contract, but copy-paste get methods:
<details>
  <summary>Full func code</summary>
    
  ```
() recv_internal(int msg_value, cell in_msg_cell, slice in_msg) impure {
}

() recv_external(slice in_msg) impure { 
  accept_message();
  set_data(begin_cell().store_uint(0, 32).store_uint(1, 32).store_uint(2, 32).store_uint(3, 32).store_uint(4, 32).store_uint(5, 32).store_uint(6, 32).store_uint(7, 32).store_uint(8, 32).store_uint(9, 32).end_cell());
}

int m0() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m1() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m2() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m3() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m4() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m5() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m6() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m7() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m8() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m9() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}

int m10() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m11() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m12() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m13() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m14() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m15() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m16() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m17() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m18() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
int m19() method_id {
  var ds = get_data().begin_parse();
  var (val0, val1, val2, val3, val4, val5, val6, val7, val8, val9) = (ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32),ds~load_uint(32));
  return val9;
}
  ```
  
</details>

After that, we do the same initiation (but with long contract code) - send gramms, send the external query. And... The contract stops to initiate.  
```
getaccount 0QA7mWNW2HnyeicOvwXlfzkHQW0JPFMxHoZ96amVV-oWtihw
...
    state:account_uninit))
...
```

After some research, the average code limit is ~400 TVM instructions. It is an extremely low amount for the contract with some business logic. For example, compiled `elector-code.fc` is larger than 2000 TVM codes.

So, how can we avoid that restriction?

**Comment by ton-blockchain** [comment]
The broadcast size for external messages has been raised to 16 KiB. Please check that large smart contracts can be deployed now, at least through full nodes (lite servers) that have been upgraded.

**Comment by qpIlIpp** [comment]
It seems like now it works. Thanks!
